src/AzureAuth.Test/ExceptionListToStringConverterTest.cs

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using FluentAssertions;
+    using Microsoft.Identity.Client;
+    using NUnit.Framework;
+
+    public class ExceptionListToStringConverterTest
+    {
+        [Test]
+        public void EmptyExceptionList()
+        {
+            IEnumerable<Exception> exceptions = new List<Exception>();
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+
+            // Assert
+            result.Should().BeNull();

```

Comment:
Rather than `null` - I think it will be helpful to have the empty set be represented by the string representation of the empty list i.e. `"[]"`. The reason for this is consistency in the data when querying. If the value is null - then they key I think is actually not included on the event which will make querying harder. If it's an empty list, this will make querying easier i think, because we can do list operations on `error_messages` and not worry about it existing or not. For instance we can easily query for where `error_messages` has a count of 0, or 1, etc. Or search where error_messages contains one with some text, etc.